include/multigrid_pclomp/multigrid_ndt_omp.h

```diff
@@ -380,9 +380,7 @@ class MultiGridNormalDistributionsTransform : public pcl::Registration<PointSour
    * \param[in] x point from the input cloud
    * \param[in] compute_hessian flag to calculate hessian, unnecessary for step calculation.
    */
-  void computePointDerivatives(Eigen::Vector3d &x, Eigen::Matrix<double, 3, 6> &point_gradient, Eigen::Matrix<double, 18, 6> &point_hessian, bool compute_hessian = true) const;
-
-  void computePointDerivatives(Eigen::Vector3d &x, Eigen::Matrix<float, 4, 6> &point_gradient, Eigen::Matrix<float, 24, 6> &point_hessian, bool compute_hessian = true) const;
```

Comment:
@SakodaShintaro Yes, actually the use of <double, 18, 6> and <double, 3, 6> matrices is from the old PCL implementation, where the columns of point's angular jacobian and hessian (j_ang_ and h_ang_) are defined as individual variables of NDT.
Currently those matrices in PCL are simplified as Eigen::Matrix variables so the computation looks less complicated, but the point gradient and point hessian are still double-type.
Probably the author of ndt_omp found that using float instead of double is faster than using double, while the accuracy is not affected much. So ndt_omp's author decided to use float matrices for point_gradient and point_hessian. As a result, we have a mix-up, where two methods perform the same functionality on different data type.